src/index.ts

```diff
@@ -399,80 +428,179 @@ export async function pressHomeButton(host) {
+        return new Promise<any>(function (resolve, reject) {
+            request.post(requestOptions, function (err, resp, body) {
+                if (err) {
+                    return reject(err);
+                }
+                return resolve({ response: resp, body: body });
             });
         }).then(function (results) {
             let error: any;
+            if (!results || !results.response || typeof results.body !== 'string') {
+                error = new Error('Invalid response');
+                error.results = results;
+                return Q.reject(error);
+            }
+
             if (options.failOnCompileError) {
-                if (results && results.body && results.body.indexOf('Install Failure: Compilation Failed.') > -1) {
+                if (results.body.indexOf('Install Failure: Compilation Failed.') > -1) {
                     error = new Error('Compile error');
+                    console.log(results.body);
+
                     error.results = results;
                     return Q.reject(error);
                 }
             }
-            if (results && results.response && results.response.statusCode === 200) {
+
+            if (results.response.statusCode === 200) {
                 if (results.body.indexOf('Identical to previous version -- not replacing.') > -1) {
                     return { message: 'Identical to previous version -- not replacing', results: results };
                 }
                 return { message: 'Successful deploy', results: results };
-            } else if (results && results.response) {
+            } else {
                 if (results.response.statusCode === 401) {
                     error = new Error('Unauthorized. Please verify username and password for target Roku.');
                 } else {
                     error = new Error('Error, statusCode other than 200: ' + results.response.statusCode);
                 }
                 error.results = results;
                 return Q.reject(error);
-            } else {
+            }
+        });
+    });
+}
+
+/**
+ * Sign a pre-existing package using Roku and return path to retrieve it
+ * @param options
+ */
+export async function signExistingPackage(options: RokuDeployOptions): Promise<string> {
+    options = getOptions(options);
+    if (!options.signingPassword) {
+        throw new Error('Must supply signingPassword');
+    }
+
+    let manifestPath = path.join(getStagingFolderPath(options), 'manifest');
+    let parsedManifest = await parseManifest(manifestPath);
+    let appName = parsedManifest.title + '/' + parsedManifest.major_version + '.' + parsedManifest.minor_version;
+
+    let requestOptions = generateBaseRequestOptions('plugin_package', options);
+    requestOptions.formData = {
+        mysubmit: 'Package',
+        pkg_time: (new Date()).getTime(),
+        passwd: options.signingPassword,
+        app_name: appName,
+    };
+
+    return Promise.all([]).then(function () {
```

Comment:
You can remove this outer Promise.all. I was using that to start a promise chain certain places. However, in this instance, you start a promise chain directly underneal on the next line...so just use that.